saleor/order/fetch.py

```diff
@@ -127,12 +123,22 @@ def fetch_draft_order_lines_info(
             DraftOrderLineInfo(
                 line=line,
                 variant=variant,
+                product=product,
+                collections=list(product.collections.all()) if product else [],
                 channel_listing=variant_channel_listing,
                 discounts=list(line.discounts.all()),
                 rules_info=rules_info,
                 channel=channel,
+                voucher=None,
+                voucher_code=None,
             )
         )
+    voucher = order.voucher
+    if voucher and (
```

Comment:
In `saleor.order.calculations.fetch_order_prices_if_expired` we also call `saleor.order.calculations._update_order_discount_for_voucher`. It creates an orderDiscount object. I don't see any condition for specfic_product or apply_once_per_order. It means, that we will propagate the discount over lines and create the additional order.discounts objects. Is it expected? Won't we add a discount twice?

Reply:
It's not the correct behaviour, we have a ticket for fixing that. I want to narrow the changes here as it's quite big right now. The `OrderDiscount` object will have `0` discount amount, so the discount won't be apply twice, but still you are right that the object shouldn't be created. We will fix that in another PR.